datetime/src/test/scala/com/fortysevendeg/scalacheck/datetime/jdk8/GenJdk8Properties.scala

```diff
@@ -132,4 +264,293 @@ object GenJdk8Properties extends Properties("Java 8 Generators") {
+        }
+      }
+    }
+
+
+  class LTGranularitiesAndPredicates extends GranularitiesAndPredicates[LocalTime] {
+    override def zeroMinutes(dt: LocalTime): Boolean = zeroSeconds(dt) && {
+      // The previous second should be in the previous hour.
+      // There are cases where half an hour has been taken out of a day,
+      // such as +58963572-10-01T02:30+11:00[Australia/Lord_Howe]
+      // One second before is 01:59:59!
+      val prevSecond = dt.plusSeconds(-1)
+      val prevHour = dt.plusHours(-1)
+
+      (prevSecond.get(HOUR_OF_DAY) == prevHour.get(HOUR_OF_DAY))
+    }
+
+    override def zeroHours(dt: LocalTime)   = zeroMinutes(dt) && dt.get(HOUR_OF_DAY) == 0
+  }
+
+  val granularitiesAndPredicatesLT: List[(Granularity[LocalTime], LocalTime => Boolean)] = {
+
+    val gp = new LTGranularitiesAndPredicates
+    import gp._
+
+    List(
+      (LocalTimeGranularity.seconds, zeroNanos _),
+      (LocalTimeGranularity.minutes, zeroSeconds _),
+      (LocalTimeGranularity.hours, zeroMinutes _),
+      (LocalTimeGranularity.days, zeroHours _)
+    )
+  }
+
+  val granularitiesAndPredicatesWithDefaultLT: List[(Granularity[LocalTime], LocalTime => Boolean)] =
+    (Granularity.identity[LocalTime], (_: LocalTime) => true) :: granularitiesAndPredicatesLT
+
+  property("genLocalTime with a granularity generates appropriate LocalTimes") =
+    forAll(Gen.oneOf(granularitiesAndPredicatesLT)) { case (granularity, predicate) =>
+
+      implicit val generatedGranularity = granularity
+
+      forAll(genLocalTime) { dt =>
+        predicate(dt) :| s"${granularity.description}: $dt"
+      }
+    }
+
+  property("arbitrary generation with a granularity generates appropriate LocalTimes") =
+    forAll(Gen.oneOf(granularitiesAndPredicatesLT)) { case (granularity, predicate) =>
+      import ArbitraryJdk8._
+
+      implicit val generatedGranularity = granularity
+
+      forAll { dt: LocalTime =>
+        predicate(dt) :| s"${granularity.description}: $dt"
+      }
+    }
+
+  // Guards against adding a duration to a datetime which cannot represent millis in a long, causing an exception.
+  private[this] def tooLargeForAddingRangesLT(dt: LocalTime, dur: Duration): Boolean = {
+    Try(dt.plusNanos(dur.getNano)).isFailure
+  }
+
+  property("genDuration can be added to any LocalTime") = forAll(genLocalTime, genDuration) { (dt, dur) =>
+    !tooLargeForAddingRangesLT(dt, dur) ==> {
+      val attempted = Try(dt.plusNanos(dur.getNano))
+      attempted.isSuccess :|  attempted.toString
+    }
+  }
+
+  property("genDateTimeWithinRange for Java 8 should generate LocalTimes between the given date and the end of the specified Duration") =
+    forAll(genLocalTime, genLocalTime, Gen.oneOf(granularitiesAndPredicatesWithDefaultLT)) { case (t1, t2, (granularity, predicate)) =>
+
+      implicit val generatedGranularity = granularity
+
+      forAll(genTimeBetween(t1, t2)) { generated =>
+
+        val resultText =
+          s"""t1:        $t1
```

Comment:
Could we use a better name than `t1` in the output? Something like "first time" (or similar) would be a bit clearer.